java/wfs-xml-query/src/main/java/com/esri/arcgisruntime/sample/wfsxmlquery/MainActivity.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2019 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.esri.arcgisruntime.sample.wfsxmlquery;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.esri.arcgisruntime.concurrent.ListenableFuture;
+import com.esri.arcgisruntime.data.FeatureQueryResult;
+import com.esri.arcgisruntime.data.ServiceFeatureTable;
+import com.esri.arcgisruntime.layers.FeatureLayer;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.ogc.wfs.OgcAxisOrder;
+import com.esri.arcgisruntime.ogc.wfs.WfsFeatureTable;
+
+public class MainActivity extends AppCompatActivity {
+
+  private static final String TAG = MainActivity.class.getSimpleName();
+
+  private MapView mMapView;
+
+  @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    mMapView = findViewById(R.id.mapView);
+
+    // create a map and set the map to display in the map view
+    ArcGISMap map = new ArcGISMap(Basemap.createNavigationVector());
+    mMapView.setMap(map);
+
+    // create the WFS feature table from URL and name
+    WfsFeatureTable statesTable = new WfsFeatureTable(getString(R.string.wfs_feature_table_url),
+        getString(R.string.wfs_feature_table_table_name));
+
+    // set the feature request mode and axis order
+    statesTable.setAxisOrder(OgcAxisOrder.NO_SWAP);
+    statesTable.setFeatureRequestMode(ServiceFeatureTable.FeatureRequestMode.MANUAL_CACHE);
+
+    try {
+      String xmlQuery = loadQueryFromAssets();
+
+      ListenableFuture<FeatureQueryResult> featureQueryResultFuture = statesTable
+          .populateFromServiceAsync(xmlQuery, true);
+
+      featureQueryResultFuture.addDoneListener(() -> {
+        // create a feature layer to visualize the table.
+        FeatureLayer statesLayer = new FeatureLayer(statesTable);
```

Comment:
I seem to have misplaced your comment about keeping this in the done listener (it might be on another related sample), but its ok to make a feature layer from an empty table and wait for as long as we want to make a call to populate the table. So I think we should keep the feature layer instantiation, and adding it to the map view outside the done listener.

My thinking behind this is, by leaving it in the listener we're implying to devs that the WfsFeatureLayer is not handled through the normal Loadable chain